cdap-unit-test/src/test/java/co/cask/cdap/flow/stream/ConnectStreamspaceFlowApp.java

```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright © 2014-2016 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.flow.stream;
+
+import co.cask.cdap.api.annotation.Batch;
+import co.cask.cdap.api.annotation.ProcessInput;
+import co.cask.cdap.api.app.AbstractApplication;
+import co.cask.cdap.api.data.stream.Stream;
+import co.cask.cdap.api.flow.AbstractFlow;
+import co.cask.cdap.api.flow.flowlet.AbstractFlowlet;
+import co.cask.cdap.api.flow.flowlet.StreamEvent;
+import com.google.common.base.Charsets;
+import com.google.common.base.Preconditions;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.Lists;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.Iterator;
+import java.util.List;
+
+/**
+ * Same as @link{TestFlowStreamIntegrationApp}, but connect to the stream in streamSpace instead of DEFAULT.
+ */
+public class ConnectStreamspaceFlowApp extends AbstractApplication {
```

Comment:
Dont use the name of the namespace from the program in class name. Call it something like: TestFlowStreamIntegrationAcrossNSApp